superset/commands/database/sync_permissions.py

```diff
@@ -192,15 +192,26 @@ def sync_database_permissions(self) -> None:
             if self.old_db_connection_name != self.db_connection.database_name:
                 self._rename_database_in_permissions(catalog, schemas)
 
-    def _get_catalog_names(self) -> set[str]:
+    def _get_catalog_names(self) -> set[str | None]:
         """
         Helper method to load catalogs.
         """
         try:
-            return self.db_connection.get_all_catalog_names(
-                force=True,
-                ssh_tunnel=self.db_connection_ssh_tunnel,
-            )
+            # Adding permissions to all catalogs (and all their schemas) can take a long
+            # time (minutes, while importing a chart, eg). If the database does not
+            # support cross-catalog queries (like RDS or Postgres), and the
```

Comment:
Based on @arafoperata's feedback, we probably want to say "like Postgres" as it's not tied to RDS